### PR TITLE
delete PullLogs

### PR DESCRIPTION
**Pull request by xzhangxian1008**
- delete all PullLogs related codes
- add codes to handle the response of PushLogs

### CODE REVIEW

File: src/rpc/services.h [hunk 1]
```diff
@@ -25,10 +25,6 @@ class NodeService {
     virtual void HandleRequestHeartbeat(RpcConn conn, int32_t term_id,
                                         std::string node_id_binary) = 0;
 
-    virtual void HandleRequestPullLogs(RpcConn conn, bool result,
-                                       std::string node_id_binary,
-                                       int64_t next_log_index) = 0;
-
     virtual void HandleRequestPushLogs(RpcConn conn, int64_t committed_log_index,
```
Comment (jovany-wang) [inline src/rpc/services.h:28]: PushLogs has no return value. How does the leader know the result or the committed index of this RPC request?

Before this PR, PullLogs can tell the leader the information.

Comment (xzhangxian1008) [reply src/rpc/services.h:28]: > PushLogs has no return value. How does the leader know the result or the committed index of this RPC request?
> 
> Before this PR, PullLogs can tell the leader the information.

In HandleRequestPushLogs, followers should response with a success flag which will let leader knows if this RPC request succeeds.

File: src/log_manager/leader_log_manager.cc [hunk 2]
```diff
@@ -109,18 +103,33 @@ void LeaderLogManager::DoPushLogs() {
...
+                msgpack::object obj = oh.get();
+                obj.convert(response);
+
+                bool success = response.get<0>();
+                int64_t resp_last_log_idx = response.get<1>();
+
+                std::lock_guard<std::mutex> lock(this->mutex_);
+
+                auto iter = this->next_log_indexes_.find(follower_node_id);
+                if (success) {
```
Comment (xzhangxian1008) [inline src/log_manager/leader_log_manager.cc:128]: Here handle the success of a RPC request

File: src/rpc/services.h (hunk 1 above)
Comment (jovany-wang) [reply src/rpc/services.h:28]: I don't think it does work. Actually, handling push logs requests in follower is async operation. So the return value `success` doesn't indicates if the follower has consumed the requests.

**Review by jovany-wang — APPROVED** [review]
Thanks. LGTM.

**Pull request merged**